build-test.sh

```diff
@@ -26,40 +26,47 @@ exit_err_cleanup() {
 }
 
 ###############################################################################
-# cleanup_test_containers removes test containers.
+# cleanup_test_containers logs massages and calls the the subcall to remove
+# test containers.
 # Globals: none
 # Parameters: none
 ###############################################################################
 cleanup_test_containers() {
-  doing "Removing test containers"
+  log_cmd "Removing test containers" cleanup_test_containers_subcall
+}
+
+###############################################################################
+# cleanup_test_containers_subcall removes test containers without logs.
+# Globals: none
+# Parameters: none
+###############################################################################
+cleanup_test_containers_subcall() {
   CONTAINERS=$(docker ps -a -q --filter "name=sia-ant-farm-test-container")
   if [ ! -z $CONTAINERS ]; then
     docker stop $CONTAINERS
     docker rm   $CONTAINERS
   fi
-  finished
 }
 
 ###############################################################################
-# doing sets and echoes the given message.
-# Globals:
-# - DOING_MSG: variable to store the message for finished() function
+# log_cmd echoes the starting message, executes the command and echoes the
+# success message.
+# Globals: none
 # Parameters:
-# - $1: The message to be set and to be echoed
+# - $1: Log message
+# - the rest: Command with parameters to be executed
 ###############################################################################
-doing() {
-  DOING_MSG=$1
-  echo "$DOING_MSG..."
-}
+log_cmd() {
+  # Get and echo log message
+	local MSG=$1
+	echo "Starting: $MSG..."
 
-###############################################################################
-# finished echoes that the message finished.
-# Globals:
-# - DOING_MSG: variable to store the message from doing() function
-# Parameters: none
-###############################################################################
-finished() {
-  echo "$DOING_MSG finished successfully"
+  # Execute the command
+  shift 1
+	"$@"
+
+  # Log success
+	echo "Success: $MSG"
```

Comment:
What happens if the command fails?

Reply:
There are settings:
```bash
## Exit immediately on error.
set -e

# Set function to be called on exit
trap 'exit_err_cleanup $?' EXIT
```

If there is an error, the script is interrupted and `exit_err_cleanup` function catches an error and exits.

The `echo "Success: $MSG"` is not executed on error.